benchmarks/run_benchmarks.py

```diff
@@ -86,34 +86,34 @@
         'euler_mod.py',
         ['euler_humps_test', 'humps_fun'],
         'import numpy as np; tspan = np.array([0.,2.]); y0 = np.array([humps_fun(0.0)]);',
-        't, y = euler_humps_test(tspan, y0, 10000)'),
+        't, y = euler_humps_test(tspan, y0, 1000000)'),
     TestInfo('Midpoint Explicit',
         'midpoint_explicit_mod.py',
         ['midpoint_explicit_humps_test', 'humps_fun'],
         'import numpy as np; tspan = np.array([0.,2.]); y0 = np.array([humps_fun(0.0)]);',
-        't, y = midpoint_explicit_humps_test(tspan, y0, 10000)'),
+        't, y = midpoint_explicit_humps_test(tspan, y0, 1000000)'),
     TestInfo('Midpoint Fixed',
         'midpoint_fixed_mod.py',
         ['midpoint_fixed_humps_test', 'humps_fun'],
         'import numpy as np; tspan = np.array([0.,2.]); y0 = np.array([humps_fun(0.0)]);',
-        't, y = midpoint_fixed_humps_test(tspan, y0, 10000)'),
+        't, y = midpoint_fixed_humps_test(tspan, y0, 1000000)'),
     TestInfo('RK4',
         'rk4_mod.py',
         ['rk4_humps_test', 'humps_fun'],
         'import numpy as np; tspan = np.array([0.,2.]); y0 = np.array([humps_fun(0.0)]);',
-        't, y = rk4_humps_test(tspan, y0, 10000)'),
+        't, y = rk4_humps_test(tspan, y0, 1000000)'),
     TestInfo('FD - L Convection',
         'linearconv_1d_mod.py',
         ['linearconv_1d'],
-        '''import numpy as np; nx=2001; nt=2000; c=1.; dt=0.0003;
+        '''import numpy as np; nx=3001; nt=3000; c=1.; dt=0.000000003;
         dx = 2 / (nx-1);
         u0 = np.ones(nx);
         u0[int(.5 / dx):int(1 / dx + 1)] = 2;''',
         'u = linearconv_1d(u0, nt, dt, dx, c)'),
     TestInfo('FD - NL Convection',
         'nonlinearconv_1d_mod.py',
         ['nonlinearconv_1d'],
-        '''import numpy as np; nx = 2001; nt=2000; c=1.; dt=0.00035;
+        '''import numpy as np; nx = 3001; nt=3000; c=1.; dt=0.000000003;
```

Comment:
Not sure why you would want to decrease the time step `dt` so drastically. All we usually need is that a Courant condition is satisfied, which is roughly `dt < dx / c` (and I think that `dx = 1 / nx` in our case, so the condition becomes `dt < 1 / nx`).

Reply:
I just tweaked the input to make the test slower without taking into the consideration its theoritical significance. I will revert these changes

Reply:
Does `dt` effect the runtime?

Reply:
if one increases nx and nt, dx gets larger and the values get too large too quickly so dt needs to be decreased to keep the values from blowing up

Reply:
I've just realized that this is the non-linear convection case, so the variable `c` is not needed at all.

Reply:
Just to clarify, `nx` is the number of points in the domain of length `2`, so the cell size is `dx = 2 / (nx - 1)`. The propagation speed for a the non-linear convection case is the solution `u` itself, which has a maximum of 2 in the initial conditions. The explicit Euler time stepping scheme must satisfy the stability limit `dt < dx / max(u)`(the Courant condition mentioned above). If the time step size `dt` is chosen too large the scheme is unstable, this is why you observe a growth in the amplitude of the solution.

Reply:
By the way, the model that is solved in this example is the 1D inviscid Burgers' equation, which can develop shocks in the solution after a certain amount of time (see [Wikipedia](https://en.wikipedia.org/wiki/Burgers%27_equation) for example). This should not be a problem as we are using one-sided finite differences (a.k.a. "upwinding") to compute the derivative `du/dx`.

Reply:
Thank you very much for the clarification!